presto-main/src/main/java/com/facebook/presto/sql/planner/LiteralEncoder.java

```diff
@@ -242,6 +243,9 @@ public Expression toExpression(Object object, Type type)
 
     public static boolean isSupportedLiteralType(Type type)
     {
+        if (type instanceof FunctionType) {
```

Comment:
Add a comment to this

Reply:
Another way is to catch `UnsupportedOperationException` from `getJavaType` but I prefer to list it out explicitly instead.